### PR TITLE
Dynamic Context for Suspened methods

### PR DESCRIPTION
**Pull request by Filmaluco**
updated: suspended fun context

### CODE REVIEW

File: example/src/main/java/org/blockstack/android/MainActivity.kt [hunk 1]
```diff
@@ -208,23 +198,21 @@ class MainActivity : AppCompatActivity() {
...
+
                 }
             }
         }
 
         getAppBucketUrlButton.setOnClickListener {
             getAppBucketUrlText.text = "Getting url ..."
             val userData = blockstackSession().loadUserData()
-            lifecycleScope.launch(Dispatchers.Main) {
+            lifecycleScope.launch() {
```
Comment (sdsantos) [inline example/src/main/java/org/blockstack/android/MainActivity.kt]: You probably can leave out the `()`.

File: example/src/main/java/org/blockstack/android/MainActivity.kt [hunk 2]
```diff
@@ -283,13 +270,8 @@ class MainActivity : AppCompatActivity() {
         }
 
         if (intent?.action == Intent.ACTION_VIEW) {
-            lifecycleScope.launch(Dispatchers.Main) {
-                val response = BlockstackConnect.handleAuthResponse(intent)
-                if (response.value != null) {
-                    runOnUiThread {
-                        onSignIn(response.value!!)
-                    }
-                }
```
Comment (sdsantos) [inline example/src/main/java/org/blockstack/android/MainActivity.kt:292]: Isn't this code needed?

**Review by sdsantos — CHANGES_REQUESTED** [review]

File: example/src/main/java/org/blockstack/android/MainActivity.kt (hunk 2 above)
Comment (Filmaluco) [reply example/src/main/java/org/blockstack/android/MainActivity.kt:292]: no longer, we were using connect to both Connect and Browser, that could result in a few issues with Browser. The generic solution is "better" :)

File: example/src/main/java/org/blockstack/android/MainActivity.kt [hunk 3]
```diff
@@ -153,12 +153,14 @@ class MainActivity : AppCompatActivity() {
             val bitMapData = stream.toByteArray()
 
             val options = PutFileOptions(false)
-            lifecycleScope.launch() {
+            lifecycleScope.launch {
                 val readURLResult = blockstackSession().putFile(imageFileName, bitMapData, options)
                 if (readURLResult.hasValue) {
                     val readURL = readURLResult.value!!
                     Log.d(TAG, "File stored at: $readURL")
+                    runOnUiThread {
```
Comment (sdsantos) [inline example/src/main/java/org/blockstack/android/MainActivity.kt]: Not needed.

**Review by sdsantos — CHANGES_REQUESTED** [review]

**Review by sdsantos — APPROVED** [review]

**Pull request closed without merging**